src/rgen/Microsoft.Macios.Generator/TabbedStringBuilder.cs

```diff
@@ -263,6 +265,34 @@ public TabbedStringBuilder CreateBlock (string line, bool block)
 		return new TabbedStringBuilder (sb, tabCount, block);
 	}
 
+	/// <summary>
+	/// Concatenate an array of lines as a single block. The generated code is similar to the following
+	///
+	/// <code>
+	/// using (var nsobject = Create ())
+	/// using (var nsstring = Create ())
+	/// using (var x = new NSString ()) {
+	///    // your code goes here.
+	/// }
+	/// </code>
+	/// </summary>
+	/// <param name="lines">The lines to concatenate.</param>
+	/// <param name="block">True if the block should use braces, false otherwise.</param>
+	/// <returns></returns>
+	public TabbedStringBuilder CreateBlock (IEnumerable<string> lines, bool block)
```

Comment:
why aren't you using https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.infrastructure.indentedstringbuilder?view=efcore-9.0

in XamlSourceGen, we use that, and have extension methods for creating blocks (using Disposable as well)

Reply:
Because we did not know about it! I'll close this PR and will see how we can move to that one.